include/experimental/__p0009_bits/mdspan.hpp

```diff
@@ -319,31 +395,26 @@ mdspan(ElementType*, SizeTypes...)
 
 MDSPAN_TEMPLATE_REQUIRES(
   class Pointer,
-  (!_MDSPAN_TRAIT(is_array, Pointer))
+  (_MDSPAN_TRAIT(is_pointer, std::remove_reference_t<Pointer>))
 )
-mdspan(const Pointer&) -> mdspan<std::remove_pointer_t<Pointer>, extents<size_t>>;
+mdspan(Pointer&&) -> mdspan<std::remove_pointer_t<std::remove_reference_t<Pointer>>, extents<size_t>>;
```

Comment:
Oh wait, we wanted `const Pointer&` still, right?

Reply:
yeah it doesn't matter really, Pointer&& does work here too. But I think people are getting worked up over stuff which doesn't matter. Since this is ONLY the deduction guide. I.e. what actually happens is determined by the constructor. This thing is only used to figure out template parameters and then a constructor is called ...

Reply:
Yup, I agree -- I saw the e-mail list discussion.